lib/web_ui/lib/src/engine/text/canvas_paragraph.dart

```diff
@@ -254,6 +255,23 @@ class CanvasParagraph implements ui.Paragraph {
   List<EngineLineMetrics> computeLineMetrics() {
     return lines.map((ParagraphLine line) => line.lineMetrics).toList();
   }
+
+  bool _disposed = false;
+
+  @override
+  void dispose() {
+    spans.clear();
+    plainText = '';
+    _disposed = true;
+  }
+
+  @override
+  bool get debugDisposed {
+    if (assertionsEnabled) {
+      return _disposed;
+    }
+    throw StateError('Vertices.debugDisposed is only avialalbe when asserts are enabled.');
```

Comment:
And this should be CanvasParagraph.

Reply:
Ditto - the user facing object is `ui.Paragraph`, which is abstract in the web implementation.

Reply:
Ah I see, thanks for the correction.